### DISCUSSION TITLE
How to customise the input field

### DISCUSSION THREAD

**Discussion by mattg1995** [flyerhq/flutter_chat_ui#152] [Q&A] [answered]
Hi all,

I've read the documentation several times and tried to figure this out but I am struggling to find it.

How do you customise the input field? My main concern is the persistent white outline border, is there any advice for this.

Thank you for the great package

**Comment by demchenkoalex** [comment]
Hey, hmm, white outline border? Don't remember having it.. We may lack some input field customisation, usually people did PRs adding specific customisable parts they needed, and I merged all of them. Will you be able to do it for your use case?

**Comment by leeyisoft** [comment]
You can borrow 509 lines of code : https://github.com/flyerhq/flutter_chat_ui/discussions/152#discussion-3675919

**Reply by leeyisoft** [reply]
https://github.com/imboy-pub/imboy-flutter/blob/main/lib/page/chat/chat_view.dart   customBottomWidget: ChatInput/4

https://github.com/imboy-pub/imboy-flutter/blob/main/lib/page/chat/widget/chat_input.dart

**Comment by demchenkoalex** [comment, marked as answer]
full customisation is part of v2 https://github.com/flyerhq/flutter_chat_ui/issues/646
